src/NServiceBus.RabbitMQ/TaskEx.cs

```diff
@@ -1,10 +1,20 @@
 namespace NServiceBus.Transport.RabbitMQ
 {
+    using System;
+    using System.Threading;
     using System.Threading.Tasks;
 
     static class TaskEx
     {
         //TODO: remove when we update to 4.6 and can use Task.CompletedTask
         public static readonly Task CompletedTask = Task.FromResult(0);
+
+        public static Task StartNew(object state, Action<object> action) => StartNew(state, action, TaskScheduler.Default);
+
+        public static Task StartNew(object state, Action<object> action, TaskScheduler scheduler) => Task.Factory.StartNew(action, state, CancellationToken.None, TaskCreationOptions.DenyChildAttach, scheduler);
+
+        public static Task StartNew(object state, Func<object, Task> function) => StartNew(state, function, TaskScheduler.Default);
+
+        public static Task StartNew(object state, Func<object, Task> function, TaskScheduler scheduler) => Task.Factory.StartNew(function, state, CancellationToken.None, TaskCreationOptions.DenyChildAttach, scheduler).Unwrap();
```

Comment:
While the generic version does make the casting nicer, it has a closure (over `action`) so it reintroduces a per-call allocation.